### PR TITLE
Adding cumsum_

### PR DESCRIPTION
**Pull request by soma2000-lang**
#10231

### CODE REVIEW

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hello,

Thanks for submitting this PR,

It seems that the test is failing for this test, it needs to be passing before we can merge :)

Thanks,
James

**Comment by soma2000-lang** [comment]
@jkeane508 the testa are failing due to files not modified by me.

**Comment by jkeane508** [comment]
> @jkeane508 the tests are failing due to files not modified by me.

Hey,

This [test](https://github.com/unifyai/ivy/actions/runs/4062507101/jobs/6993645160#step:3:260) indicates that the test in this PR is failing :)

But as far as lint errors go, yes I agree they are for different files not relevant to this PR, so no need to worry about them!

Happy Coding,
James

**Comment by soma2000-lang** [comment]
@jkeane508  Can you check now.

**Review by jkeane508 — CHANGES_REQUESTED** [review]
Hey,

So looks like the test is failing due to the input dtype and output dtype not being the same in the tests. You can see that [here](https://github.com/unifyai/ivy/actions/runs/4108402481/jobs/7089053736)

Feel free to ask any questions :)

**Comment by soma2000-lang** [comment]
@jkeane508  I think cumsum has been implemented wrong.Also I did a mistake of writing self.ivy_array instead of self._ivy_array.In pr #10487.I corrected that ,but then also the tests are failing.Please continue on  PR #10487.

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 1]
```diff
@@ -3588,6 +3588,49 @@ def test_torch_instance_cumsum(
...
+    ),
+    dim=helpers.get_axis(
+        shape=st.shared(helpers.get_shape(), key="shape"),
+        allow_neg=True,
+        force_int=True,
+    ),
+    dtypes=_dtypes(),
+)
+def test_torch_instance_cumsum_(
+    dtype_value,
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3608]: here you are generating a dtype

File: ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py [hunk 2]
```diff
@@ -3588,6 +3588,49 @@ def test_torch_instance_cumsum(
...
+    dim,
+    dtypes,
```
Comment (jkeane508) [inline ivy_tests/test_ivy/test_frontends/test_torch/test_tensor.py:3610]: and also here

**Review by jkeane508 — CHANGES_REQUESTED** [review]
The issue seems to be with your test actually, you're using two different dtypes in the generation which is why the test is failing.

**Comment by soma2000-lang** [comment]
@jkeane508 I have added the  changes in PR https://github.com/unifyai/ivy/pull/10487.But still the tests fail.

**Pull request closed without merging**